### ISSUE TITLE
Failed to remove read-only file (during platform build)

### ISSUE THREAD

**Issue by GoZOo** [platformsh/platformsh-cli#453]
Some times (very often since i work on a new project since few days), my platform build fail after downloading every modules of my project.make.

```
[error]  Base path /home/vagrant/data/PROJECT/.platform/local/builds/default/public already exists. 

                                                                                                                                                                                                         
  [Exception]                                                                                                                                                                                            
  The command failed with the exit code: 1                                                                                                                                                               
                                                                                                                                                                                                         
  Full command: '/usr/local/bin/drush' 'make' '/home/vagrant/data/PROJECT/project-core.make' '/home/vagrant/data/PROJECT/.platform/local/builds/default/public' '--yes' '--cache-duration-releasexml=  
  300' '--concurrency=4' 
```

Then i remove folders build-archives, default.bak and some times tmp-public and plaform build works.

platform-cli is currently 3.1.1, but i had same issue with 3.0.7
PHP-cli is PHP 7.0.4-7ubuntu2.1, built with ubuntu 16.04

BTW, which is the purpose of default.bak ? Seems this could be removed.

**Comment by pjcdawkins** [comment]
The CLI creates a backup of your previous build directory each time `platform build` is run. It's a little bit of insurance against deleting data, in case the user hasn't realised the `platform build` will completely replace their previous build. I'd like to keep that insurance.

It moves the directory like this:
```
.platform/local/builds/default -> .platform/local/builds/default.bak
```

Of course that means removing the previous `default.bak` folder.

If you run in verbose mode (`platform build -v`) you'll see the PHP error: most probably, something exists in `default.bak` that has file permissions that does not let you delete it, such as possibly Drupal's `tmp` folder.

This means that `default` is never moved, and thus `public` already exists.

I think the solution would be to save this 'backup' as an archive - then it can easily be deleted. I'll have a go at that...

**Comment by pjcdawkins** [comment]
This should now be fixed in 3.1.2 (use `platform up` to update)

**Comment by GoZOo** [comment]
I still have an issue building a project with 3.1.2: 
```Failed to remove directory /home/vagrant/data/PROJECT/.platform/local/builds/default```

I guess it's caused by settings.php file permissions.

**Comment by pjcdawkins** [comment]
Right... at least the error is clearer. Add -v for more detail

**Comment by GoZOo** [comment]
```
Running command: 'tar' '-czp' '-C/home/vagrant/data/PROJECT/.platform/local/builds/default' '-f/home/vagrant/data/PROJECT/.platform/local/builds/default-old.tar.gz' '.'

Warning: Failed to remove file "/home/vagrant/data/PROJECT/.platform/local/builds/default/public/sites/default/settings.php". in phar:///home/vagrant/.platformsh/bin/platform/src/Helper/FilesystemHelper.php on line 80

Failed to remove directory /home/vagrant/data/PROJECT/.platform/local/builds/default
```

**Comment by pjcdawkins** [comment]
Possibly the `remove()` method could attempt to recursively `chmod` read-only files (like `settings.php`) before attempting to delete them. That sounds slow. Or it could retry with a `chmod` just if it encounters an error. I could investigate...

**Comment by pjcdawkins** [comment]
Should be fixed in `v3.3.0`
